client/components/jetpack/connection-health/use-check-jetpack-connection-health.tsx

```diff
@@ -0,0 +1,38 @@
+import { useQuery } from '@tanstack/react-query';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import wp from 'calypso/lib/wp';
+import { useSelector } from 'calypso/state';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export const JETPACK_CONNECTION_HEALTH_QUERY_KEY = 'jetpack-connection-health';
+
+export interface JetpackConnectionHealth {
+	is_healthy: boolean;
+	error: string;
+}
+
+export const useCheckJetpackConnectionHealth = ( siteId: number | null ) => {
+	const query = useQuery< JetpackConnectionHealth, unknown, JetpackConnectionHealth >( {
```

Comment:
I wonder if the data you manage here can be another reducer alongside the existing Redux code related to Jetpack connection health. Refetching should be more straightforward because you stay inside one layer.

Reply:
Agreed. We should definitely continue using the same layer. If you really feel the Jetpack state needs to use React Query, you might want to refactor it first. I think it would be easier to just add another Redux subtree and that will make the integration and data exchange much easier.